### PR TITLE
update docs for tiflash pipeline model

### PR DESCRIPTION
**Pull request by SeaRise**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
- pipeline model supports S3 and join spill
- Variable level changed to global
- pipeline model is enabled by default
- rename `tidb_enable_tiflash_pipeline_model` to `tiflash_enable_pipeline_model`

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/14464
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash/tiflash-pipeline-model.md [hunk 1]
```diff
@@ -43,32 +23,22 @@ SHOW GLOBAL VARIABLES LIKE 'tidb_enable_tiflash_pipeline_model';
...
-    ```sql
-    SET SESSION tidb_enable_tiflash_pipeline_model=ON;
-    ```
+变量 `tidb_enable_tiflash_pipeline_model` 支持 global 级别的修改。
 
 - 如果需要在 global 级别开启 TiFlash Pipeline Model，可以通过以下语句设置：
 
     ```sql
     SET GLOBAL tidb_enable_tiflash_pipeline_model=ON;
     ```
```
Comment (Oreoxmt) [inline tiflash/tiflash-pipeline-model.md:37]:
````suggestion
你可以通过以下语句在 global 级别开启 TiFlash Pipeline Model：

```sql
SET GLOBAL tidb_enable_tiflash_pipeline_model=ON;
```
````

File: tiflash/tiflash-pipeline-model.md [hunk 2]
```diff
@@ -43,32 +23,22 @@ SHOW GLOBAL VARIABLES LIKE 'tidb_enable_tiflash_pipeline_model';
...
 
-    设置 global 级别后，新建的会话中 session 和 global 级别的 `tidb_enable_tiflash_pipeline_model` 都将默认启用新值。
+    设置 global 级别后，新建的会话中 `tidb_enable_tiflash_pipeline_model` 将默认启用新值。
```
Comment (Oreoxmt) [inline tiflash/tiflash-pipeline-model.md:38]:
```suggestion
设置 global 级别后，新建的会话中 `tidb_enable_tiflash_pipeline_model` 将默认启用新值。
```

File: system-variables.md [hunk 3]
```diff
@@ -2012,22 +2012,14 @@ Query OK, 0 rows affected (0.09 sec)
...
 
-- 作用域：SESSION | GLOBAL
+- 作用域：GLOBAL
 - 是否持久化到集群：是
 - 类型：布尔型
-- 默认值：`OFF`
+- 默认值：`ON`
 - 这个变量用来控制是否启用 TiFlash 新的执行模型 [Pipeline Model](/tiflash/tiflash-pipeline-model.md)。
 - 当设置该变量为 `OFF` 关闭 TiFlash Pipeline Model 时，下推到 TiFlash 的查询会使用 TiFlash 原有的执行模型 Stream Model 来执行。
 - 当设置该变量为 `ON` 开启 TiFlash Pipeline Model 时，下推到 TiFlash 的查询会使用 TiFlash 新的执行模型 Pipeline Model 来执行。
```
Comment (Oreoxmt) [inline system-variables.md:2021]:
```suggestion
- 当设置该变量为 `ON` 开启 TiFlash Pipeline Model 时，下推到 TiFlash 的查询会使用 TiFlash 新的执行模型 Pipeline Model 来执行。
- 当设置该变量为 `OFF` 关闭 TiFlash Pipeline Model 时，下推到 TiFlash 的查询会使用 TiFlash 原有的执行模型 Stream Model 来执行。
```

File: tiflash/tiflash-pipeline-model.md [hunk 4]
```diff
@@ -43,31 +23,20 @@ SHOW GLOBAL VARIABLES LIKE 'tidb_enable_tiflash_pipeline_model';
...
-    设置 global 级别后，新建的会话中 session 和 global 级别的 `tidb_enable_tiflash_pipeline_model` 都将默认启用新值。
-
-如需关闭 TiFlash Pipeline Model，可以通过以下语句设置：
+你可以通过以下语句在 global 级别开启 TiFlash Pipeline Model：
 
 ```sql
-SET SESSION tidb_enable_tiflash_pipeline_model=OFF;
+SET GLOBAL tidb_enable_tiflash_pipeline_model=ON;
 ```
+设置 global 级别后，新建的会话中 `tidb_enable_tiflash_pipeline_model` 将默认启用新值。
```
Comment (Oreoxmt) [inline tiflash/tiflash-pipeline-model.md:37]:
```suggestion

设置 global 级别后，新建的会话中 `tidb_enable_tiflash_pipeline_model` 将默认启用新值。
```

**Review by Oreoxmt — APPROVED** [review]

**Comment by Oreoxmt** [comment]
@SeaRise Could you please invite a tech reviewer? Thanks!

**Review by windtalker — APPROVED** [review]
LGTM

**Review by ran-huang — APPROVED** [review]

**Review by Oreoxmt — APPROVED** [review]

File: system-variables.md [hunk 5]
```diff
@@ -2010,23 +2010,15 @@ Query OK, 0 rows affected (0.09 sec)
...
-
-> **注意：**
->
-> - TiFlash Pipeline Model 目前为实验特性，不建议在生产环境中使用。
-> - TiFlash Pipeline Model 目前不支持以下功能。当下列功能开启时，即使 `tidb_enable_tiflash_pipeline_model` 设置为 `ON`，下推到 TiFlash 的查询仍会使用原有的执行模型 Stream Model 来执行。
->
->     - [Join 算子落盘](#tidb_max_bytes_before_tiflash_external_join-从-v700-版本开始引入)
->     - [TiFlash 存算分离架构与 S3](/tiflash/tiflash-disaggregated-and-s3.md)
+- 当设置该变量为 `OFF` 关闭 TiFlash Pipeline Model 时，下推到 TiFlash 的查询会使用 TiFlash 原有的执行模型 Stream Model 来执行。
```
Comment (Oreoxmt) [inline system-variables.md:2021]:
```suggestion
- 当设置该变量为 `OFF` 关闭 TiFlash Pipeline Model 时，下推到 TiFlash 的查询会使用 TiFlash 原有的执行模型 Stream Model 来执行。
- 该变量由 `tidb_enable_tiflash_pipeline_model` 改名而来。
```

File: system-variables.md [hunk 6]
```diff
@@ -2010,23 +2010,15 @@ Query OK, 0 rows affected (0.09 sec)
 - 默认值：`OFF`
 - 这个变量用于动态地控制 TiDB 遥测功能是否开启，当前版本默认关闭 TiDB 的遥测功能。当所有 TiDB 实例都设置配置项 [`enable-telemetry`](/tidb-configuration-file.md#enable-telemetry-从-v402-版本开始引入) 为 `false` 时，将忽略该系统变量，并总是关闭 TiDB 遥测功能。参阅[遥测](/telemetry.md)了解该功能详情。
 
-### `tidb_enable_tiflash_pipeline_model` <span class="version-mark">从 v7.2.0 版本开始引入</span>
+### `tiflash_enable_pipeline_model` <span class="version-mark">从 v7.2.0 版本开始引入</span>
```
Comment (Oreoxmt) [inline system-variables.md:2013]:
```suggestion
### `tiflash_enable_pipeline_model` <span class="version-mark">从 v7.3.0 版本开始引入</span>
```

File: releases/release-7.2.0.md [hunk 7]
```diff
@@ -56,7 +56,7 @@ TiDB 版本：7.2.0
 
 * TiFlash 支持 Pipeline 执行模型（实验特性）[#6518](https://github.com/pingcap/tiflash/issues/6518) @[SeaRise](https://github.com/SeaRise)
 
-    在 v7.2.0 版本之前，TiFlash 引擎中各个任务在执行时，需要自行申请线程资源。TiFlash 引擎通过控制任务数的方式限制线程资源使用，以避免线程资源超用，但并不能完全避免此问题。因此，在 v7.2.0 中，TiFlash 引入 Pipeline 执行模型，对所有线程资源进行统一管理，并对所有任务的执行进行统一调度，充分利用线程资源，同时避免资源超用。新增系统变量 [`tidb_enable_tiflash_pipeline_model`](/system-variables.md#tidb_enable_tiflash_pipeline_model-从-v720-版本开始引入) 用于设置是否启用 Pipeline 执行模型。
+    在 v7.2.0 版本之前，TiFlash 引擎中各个任务在执行时，需要自行申请线程资源。TiFlash 引擎通过控制任务数的方式限制线程资源使用，以避免线程资源超用，但并不能完全避免此问题。因此，在 v7.2.0 中，TiFlash 引入 Pipeline 执行模型，对所有线程资源进行统一管理，并对所有任务的执行进行统一调度，充分利用线程资源，同时避免资源超用。新增系统变量 `tidb_enable_tiflash_pipeline_model` 用于设置是否启用 Pipeline 执行模型。
```
Comment (Oreoxmt) [inline releases/release-7.2.0.md:59]:
```suggestion
    在 v7.2.0 版本之前，TiFlash 引擎中各个任务在执行时，需要自行申请线程资源。TiFlash 引擎通过控制任务数的方式限制线程资源使用，以避免线程资源超用，但并不能完全避免此问题。因此，在 v7.2.0 中，TiFlash 引入 Pipeline 执行模型，对所有线程资源进行统一管理，并对所有任务的执行进行统一调度，充分利用线程资源，同时避免资源超用。新增系统变量 [`tidb_enable_tiflash_pipeline_model`](https://docs.pingcap.com/zh/tidb/v7.2/system-variables#tidb_enable_tiflash_pipeline_model-从-v720-版本开始引入) 用于设置是否启用 Pipeline 执行模型。
```

File: releases/release-7.2.0.md [hunk 8]
```diff
@@ -180,7 +180,7 @@ TiDB 版本：7.2.0
 | [`tidb_analyze_skip_column_types`](/system-variables.md#tidb_analyze_skip_column_types-从-v720-版本开始引入) | 新增 | 这个变量表示在执行 `ANALYZE` 命令收集统计信息时，跳过哪些类型的列的统计信息收集。该变量仅适用于 [`tidb_analyze_version = 2`](/system-variables.md#tidb_analyze_version-从-v510-版本开始引入) 的情况。使用 `ANALYZE TABLE t COLUMNS c1, ..., cn` 语法时，如果指定的列的类型在 `tidb_analyze_skip_column_types` 中，则不会收集该列的统计信息。 |
 | [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint-从-v720-版本开始引入) | 新增 | 这个变量用于控制 `CHECK` 约束功能是否开启。默认值 `OFF` 表示该功能默认关闭。 |
 | [`tidb_enable_fast_table_check`](/system-variables.md#tidb_enable_fast_table_check-从-v720-版本开始引入) | 新增 | 这个变量用于控制是否使用基于校验和的方式来快速检查表中数据和索引的一致性。默认值 `ON` 表示该功能默认开启。  |
-| [`tidb_enable_tiflash_pipeline_model`](/system-variables.md#tidb_enable_tiflash_pipeline_model-从-v720-版本开始引入) | 新增 | 这个变量用来控制是否启用 TiFlash 新的执行模型 [Pipeline Model](/tiflash/tiflash-pipeline-model.md)，默认值为 `OFF`，即关闭 Pipeline Model。 |
+| `tidb_enable_tiflash_pipeline_model` | 新增 | 这个变量用来控制是否启用 TiFlash 新的执行模型 [Pipeline Model](/tiflash/tiflash-pipeline-model.md)，默认值为 `OFF`，即关闭 Pipeline Model。 |
```
Comment (Oreoxmt) [inline releases/release-7.2.0.md:183]:
```suggestion
| [`tidb_enable_tiflash_pipeline_model`](https://docs.pingcap.com/zh/tidb/v7.2/system-variables#tidb_enable_tiflash_pipeline_model-从-v720-版本开始引入) | 新增 | 这个变量用来控制是否启用 TiFlash 新的执行模型 [Pipeline Model](/tiflash/tiflash-pipeline-model.md)，默认值为 `OFF`，即关闭 Pipeline Model。 |
```

File: tiflash/tiflash-pipeline-model.md [hunk 9]
```diff
@@ -7,70 +7,40 @@ summary: 介绍 TiFlash 新的执行模型 Pipeline Model。
 
 本文介绍 TiFlash 新的执行模型 Pipeline Model。
 
-从 v7.2.0 起，TiFlash 支持新的执行模型 Pipeline Model。你可以通过修改变量 [`tidb_enable_tiflash_pipeline_model`](/system-variables.md#tidb_enable_tiflash_pipeline_model-从-v720-版本开始引入) 来控制是否启用 TiFlash Pipeline Model。
+从 v7.2.0 起，TiFlash 支持新的执行模型 Pipeline Model。你可以通过修改变量 [`tiflash_enable_pipeline_model`](/system-variables.md#tiflash_enable_pipeline_model-从-v720-版本开始引入) 来控制是否启用 TiFlash Pipeline Model。
```
Comment (Oreoxmt) [inline tiflash/tiflash-pipeline-model.md:10]:
```suggestion
从 v7.2.0 起，TiFlash 支持新的执行模型 Pipeline Model。你可以通过以下系统变量来控制是否启用 TiFlash Pipeline Model：

- v7.2.0：使用 [`tidb_enable_tiflash_pipeline_model`](https://docs.pingcap.com/zh/tidb/v7.2/system-variables#tidb_enable_tiflash_pipeline_model-从-v720-版本开始引入) 控制。
- v7.3.0 及之后版本：`tidb_enable_tiflash_pipeline_model` 重命名为 `tiflash_enable_pipeline_model`。你可以使用  [`tiflash_enable_pipeline_model`](/system-variables.md#tiflash_enable_pipeline_model-从-v720-版本开始引入) 控制。
```

**Review by Oreoxmt — APPROVED** [review]

File: tiflash/tiflash-pipeline-model.md [hunk 10]
```diff
@@ -7,70 +7,43 @@ summary: 介绍 TiFlash 新的执行模型 Pipeline Model。
 
 本文介绍 TiFlash 新的执行模型 Pipeline Model。
 
-从 v7.2.0 起，TiFlash 支持新的执行模型 Pipeline Model。你可以通过修改变量 [`tidb_enable_tiflash_pipeline_model`](/system-variables.md#tidb_enable_tiflash_pipeline_model-从-v720-版本开始引入) 来控制是否启用 TiFlash Pipeline Model。
+从 v7.2.0 起，TiFlash 支持新的执行模型 Pipeline Model。你可以通过以下系统变量来控制是否启用 TiFlash Pipeline Model：
 
-Pipeline Model 主要借鉴了 [Morsel-Driven Parallelism: A NUMA-Aware Query Evaluation Framework for the Many-Core Age](https://dl.acm.org/doi/10.1145/2588555.2610507) 这篇论文，提供了一个精细的任务调度模型，有别于传统的线程调度模型，减少了操作系统申请和调度线程的开销以及提供精细的调度机制。
+- v7.2.0：使用 [`tidb_enable_tiflash_pipeline_model`](https://docs.pingcap.com/zh/tidb/v7.2/system-variables#tidb_enable_tiflash_pipeline_model-从-v720-版本开始引入) 控制。
```
Comment (Oreoxmt) [inline tiflash/tiflash-pipeline-model.md:12]:
```suggestion
- v7.2.0 和 v7.3.0：使用 [`tidb_enable_tiflash_pipeline_model`](https://docs.pingcap.com/zh/tidb/v7.2/system-variables#tidb_enable_tiflash_pipeline_model-从-v720-版本开始引入) 控制。
```

File: system-variables.md [hunk 11]
```diff
@@ -2010,23 +2010,16 @@ Query OK, 0 rows affected (0.09 sec)
 - 默认值：`OFF`
 - 这个变量用于动态地控制 TiDB 遥测功能是否开启，当前版本默认关闭 TiDB 的遥测功能。当所有 TiDB 实例都设置配置项 [`enable-telemetry`](/tidb-configuration-file.md#enable-telemetry-从-v402-版本开始引入) 为 `false` 时，将忽略该系统变量，并总是关闭 TiDB 遥测功能。参阅[遥测](/telemetry.md)了解该功能详情。
 
-### `tidb_enable_tiflash_pipeline_model` <span class="version-mark">从 v7.2.0 版本开始引入</span>
+### `tiflash_enable_pipeline_model` <span class="version-mark">从 v7.4.0 版本开始引入</span>
```
Comment (ran-huang) [inline system-variables.md:2013]: 改名后排序也变了，麻烦字母顺序排列

**Review by ran-huang — APPROVED** [review]
rest LGTM

File: releases/release-7.2.0.md [hunk 12]
```diff
@@ -56,7 +56,7 @@ TiDB 版本：7.2.0
 
 * TiFlash 支持 Pipeline 执行模型（实验特性）[#6518](https://github.com/pingcap/tiflash/issues/6518) @[SeaRise](https://github.com/SeaRise)
 
-    在 v7.2.0 版本之前，TiFlash 引擎中各个任务在执行时，需要自行申请线程资源。TiFlash 引擎通过控制任务数的方式限制线程资源使用，以避免线程资源超用，但并不能完全避免此问题。因此，在 v7.2.0 中，TiFlash 引入 Pipeline 执行模型，对所有线程资源进行统一管理，并对所有任务的执行进行统一调度，充分利用线程资源，同时避免资源超用。新增系统变量 [`tidb_enable_tiflash_pipeline_model`](/system-variables.md#tidb_enable_tiflash_pipeline_model-从-v720-版本开始引入) 用于设置是否启用 Pipeline 执行模型。
+    在 v7.2.0 版本之前，TiFlash 引擎中各个任务在执行时，需要自行申请线程资源。TiFlash 引擎通过控制任务数的方式限制线程资源使用，以避免线程资源超用，但并不能完全避免此问题。因此，在 v7.2.0 中，TiFlash 引入 Pipeline 执行模型，对所有线程资源进行统一管理，并对所有任务的执行进行统一调度，充分利用线程资源，同时避免资源超用。新增系统变量 [`tidb_enable_tiflash_pipeline_model`](https://docs.pingcap.com/zh/tidb/v7.2/system-variables#tidb_enable_tiflash_pipeline_model-从-v720-版本开始引入) 用于设置是否启用 Pipeline 执行模型。
```
Comment (joey-yez) [inline releases/release-7.2.0.md:59]: 7.2 的 release note 是不是不需要修改了？

File: releases/release-7.2.0.md [hunk 13]
```diff
@@ -180,7 +180,7 @@ TiDB 版本：7.2.0
 | [`tidb_analyze_skip_column_types`](/system-variables.md#tidb_analyze_skip_column_types-从-v720-版本开始引入) | 新增 | 这个变量表示在执行 `ANALYZE` 命令收集统计信息时，跳过哪些类型的列的统计信息收集。该变量仅适用于 [`tidb_analyze_version = 2`](/system-variables.md#tidb_analyze_version-从-v510-版本开始引入) 的情况。使用 `ANALYZE TABLE t COLUMNS c1, ..., cn` 语法时，如果指定的列的类型在 `tidb_analyze_skip_column_types` 中，则不会收集该列的统计信息。 |
 | [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint-从-v720-版本开始引入) | 新增 | 这个变量用于控制 `CHECK` 约束功能是否开启。默认值 `OFF` 表示该功能默认关闭。 |
 | [`tidb_enable_fast_table_check`](/system-variables.md#tidb_enable_fast_table_check-从-v720-版本开始引入) | 新增 | 这个变量用于控制是否使用基于校验和的方式来快速检查表中数据和索引的一致性。默认值 `ON` 表示该功能默认开启。  |
-| [`tidb_enable_tiflash_pipeline_model`](/system-variables.md#tidb_enable_tiflash_pipeline_model-从-v720-版本开始引入) | 新增 | 这个变量用来控制是否启用 TiFlash 新的执行模型 [Pipeline Model](/tiflash/tiflash-pipeline-model.md)，默认值为 `OFF`，即关闭 Pipeline Model。 |
+| [`tidb_enable_tiflash_pipeline_model`](https://docs.pingcap.com/zh/tidb/v7.2/system-variables#tidb_enable_tiflash_pipeline_model-从-v720-版本开始引入) | 新增 | 这个变量用来控制是否启用 TiFlash 新的执行模型 [Pipeline Model](/tiflash/tiflash-pipeline-model.md)，默认值为 `OFF`，即关闭 Pipeline Model。 |
```
Comment (joey-yez) [inline releases/release-7.2.0.md:183]: remove this

File: tiflash/tiflash-pipeline-model.md [hunk 14]
```diff
@@ -7,70 +7,43 @@ summary: 介绍 TiFlash 新的执行模型 Pipeline Model。
 
 本文介绍 TiFlash 新的执行模型 Pipeline Model。
 
-从 v7.2.0 起，TiFlash 支持新的执行模型 Pipeline Model。你可以通过修改变量 [`tidb_enable_tiflash_pipeline_model`](/system-variables.md#tidb_enable_tiflash_pipeline_model-从-v720-版本开始引入) 来控制是否启用 TiFlash Pipeline Model。
+从 v7.2.0 起，TiFlash 支持新的执行模型 Pipeline Model。你可以通过以下系统变量来控制是否启用 TiFlash Pipeline Model：
```
Comment (joey-yez) [inline tiflash/tiflash-pipeline-model.md:10]:
```suggestion
从 v7.2.0 起，TiFlash 支持新的执行模型 Pipeline Model。
从 v7.4.0 起，TiFlash 正式支持 Pipeline Model 模型。Pipeline Model 和 TiFlash resource control 功能依赖度非常高，开启 TiFlash resource control 功能时，Pipeline Model 模型将自动启用。具体 TiFlash resource control 功能的启用方式，请参考对应文档。
```

Comment (joey-yez) [reply tiflash/tiflash-pipeline-model.md:10]: 需要把 tiflash resource control 开启的文档链接加到这里。

Comment (SeaRise) [reply tiflash/tiflash-pipeline-model.md]: 要不要把 pipeline 的 doc 更新合并到 https://github.com/pingcap/docs-cn/pull/15043 里面？

Comment (SeaRise) [reply tiflash/tiflash-pipeline-model.md]: 因为好像 resource control 的 doc pr 不合，这个 pr 也没法合，而且这个 doc 也是 resource control 才要改.
/cc @guo-shaoge

**Comment by guo-shaoge** [comment]
https://github.com/pingcap/docs-cn/pull/15043

**Pull request closed without merging**